### DISCUSSION TITLE
Java metrics SDK histogram being received by collector as a summary?

### DISCUSSION THREAD

**Discussion by dparis** [open-telemetry/opentelemetry-java#3569] [Q&A] [answered]
Hello!

I'm working on a trial exploration of the OpenTelemetry framework for Java. I've got some code using the 1.5.0(-alpha) version of the SDK which does the following:

1. Initializes an `OtlpGrpcMetricExporter` instance
2. Initializes a `SdkMeterProvider` instance
3. Uses both of the above to initialize an `IntervalMetricReader` instance
4. Uses the above provider to create a `Meter` and calls `histogramBuilder` to create a histogram instance
5. Calls `.record` on the histogram, passing an integer value

I have the OT Collector running with the otlp/grpc receiver configured, and the prometheus exporter configured.

After calling `.record`, I would expect to see a histogram being exported via the `/metrics` endpoint, but instead I'm seeing a summary type. I thought this might be due to converting the metric to the native prometheus format, but the debug output for the Collector seems to indicate that the metric is being received as `Summary`:

```
otel-collector_1  | Resource labels:
otel-collector_1  |      -> service.name: STRING(unknown_service:java)
otel-collector_1  |      -> telemetry.sdk.language: STRING(java)
otel-collector_1  |      -> telemetry.sdk.name: STRING(opentelemetry)
otel-collector_1  |      -> telemetry.sdk.version: STRING(1.5.0)
otel-collector_1  | InstrumentationLibraryMetrics #0
otel-collector_1  | InstrumentationLibrary test-app.java
otel-collector_1  | Metric #0
otel-collector_1  | Descriptor:
otel-collector_1  |      -> Name: test_java_histogram
otel-collector_1  |      -> Description: a histogram from java
otel-collector_1  |      -> Unit: 1
otel-collector_1  |      -> DataType: *Summary*
otel-collector_1  | SummaryDataPoints #0
otel-collector_1  | StartTimestamp: 2021-08-31 00:51:16.62543 +0000 UTC
otel-collector_1  | Timestamp: 2021-08-31 00:51:21.647018 +0000 UTC
otel-collector_1  | Count: 1
otel-collector_1  | Sum: 123.000000
otel-collector_1  | QuantileValue #0: Quantile 0.000000, Value 123.000000
otel-collector_1  | QuantileValue #1: Quantile 1.000000, Value 123.000000
```

Is this expected? A bug? Version mismatch somewhere between library and/or collector?

I know that the metrics API/SDK is alpha right now, just trying to understand if this is a misconfiguration or something on my end or not. Cheers!

**Comment by anuraaga** [comment, marked as answer]
Hi @dparis - 1.6 will report histograms as histogram, 1.5 indeed currently exports as summary. In the java agent, we have a workaround to dogfood the behavior in the 1.6 release to map the duration metrics it creates to histogram, you may want to use similar in your app for now

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java#L86

**Reply by dparis** [reply]
Perfect, thanks so much!

**Reply by dparis** [reply]
Also, GH is returning 422 error when I attempt to "Mark as answer", so that'll have to wait. But this answered my question!
